### PR TITLE
Get class endpoint

### PR DESCRIPTION
**Pull request by jolenezheng**
### Notion ticket link

[Ticket Link](https://www.notion.so/uwblueprintexecs/9cb454262ed84c6299c0682c23f92ec5?v=1076bd30711a4a01838becee05f8b5a3&p=e09a1193006f4621a54a47550cab853d)

### Implementation description

- Added a classDetail serializer
- Exposed a GET endpoint to get class details
- Added tests for the serializer and for the endpoint

### Steps to test

1. ./manage.py test
2. Can also create classes and test through the Django Admin and GET /classes endpoint

### What should reviewers focus on?

- Logic to only get active enrolments is in the right place 😵
- Test coverage
- Code cleanliness

### Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: enrolments/serializers.py [hunk 1]
```diff
@@ -31,3 +32,28 @@ class ClassListSerializer(serializers.HyperlinkedModelSerializer):
...
+        ]
+
+    def get_families(self, obj):
+        e_set = Enrolment.objects.filter(enrolled_class=obj)
+        return [
+            FamilyAttendanceSerializer(
+                enrolment.family, read_only=True, context={"request": None}
+            ).data
+            for enrolment in e_set
+            if enrolment.active == True
```
Comment (jolenezheng) [inline enrolments/serializers.py]: Added logic to only get families who have an active enrolment here. The logic for this should be in the ClassDetail serializer right?

File: enrolments/serializers.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 from rest_framework import serializers
 from registration.models import Family
 from registration.serializers import StudentSerializer
+from enrolments.models import Enrolment
 from .models import Session, Class
```
Comment (brittanylau) [inline enrolments/serializers.py]: nit: can we squish these import statements together?

File: enrolments/views.py [hunk 3]
```diff
@@ -31,3 +32,16 @@ class SessionViewSet(
     def get_classes(self, request, pk=None):
         classes = Session.objects.get(id=pk).classes
         return Response(ClassListSerializer(classes, many=True).data)
+
+
+class ClassViewSet(
+    viewsets.GenericViewSet,
+    mixins.ListModelMixin,
```
Comment (brittanylau) [inline enrolments/views.py:39]: we only need the ClassDetailSerializer for GET /classes/:id, so we shouldn't need to add a list model mixin here!

File: enrolments/serializers.py (hunk 1 above)
Comment (brittanylau) [reply enrolments/serializers.py]: yes this logic looks good to me! one nit is to move active=True to the filter on line 52, so something like this:
```
e_set = Enrolment.objects.filter(enrolled_class=obj, active=True)
```

File: enrolments/tests/views/test_class.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+        response = self.client.delete(url)
+        self.assertEqual(response.status_code, status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    def test_unauthorized(self):
+        url = reverse("classes-detail", args=[self.class1.id])
+
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
```
Comment (brittanylau) [inline enrolments/tests/views/test_class.py:102]: can we also ensure GET classes-list and PUT classes-detail are protected?

we should be able to copy/paste from [here](https://github.com/uwblueprint/project-read-backend/blob/main/registration/tests/views/test_family.py#L98-L116)!

**Review by brittanylau — APPROVED** [review]
overall lgtm!!! a few notes, feel free to merge after 🌟

**Pull request merged**